specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/file.json

```diff
@@ -0,0 +1,1099 @@
+            "type": "string",
+            "description": "Optional. Valid values are: snapshots, leased-snapshots, none. The default value is snapshots. For 'snapshots', the file share is deleted including all of its file share snapshots. If the file share contains leased-snapshots, the deletion fails. For 'leased-snapshots', the file share is deleted included all of its file share snapshots (leased/unleased). For 'none', the file share is deleted if it has no share snapshots. If the file share contains any snapshots (leased or unleased), the deletion fails."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Delete Share operation completed successfully."
+          },
+          "204": {
+            "description": "No Content -- The Share not exist."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/fileServices/default/shares/{shareName}/restore": {
+      "post": {
+        "tags": [
+          "FileShares"
+        ],
+        "operationId": "FileShares_Restore",
+        "x-ms-examples": {
+          "RestoreShares": {
+            "$ref": "./examples/FileSharesRestore.json"
+          }
+        },
+        "description": "Restore a file share within a valid retention days if share soft delete is enabled",
+        "parameters": [
+          {
+            "$ref": "./storage.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "./storage.json#/parameters/StorageAccountName"
+          },
+          {
+            "$ref": "#/parameters/ShareName"
+          },
+          {
+            "name": "deletedShare",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DeletedShare"
+            }
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- The Share is successfully restored."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/fileServices/default/shares/{shareName}/lease": {
```

Comment:
The generated .net SDK will give following failure, although the request run success.

> Microsoft.Rest.ValidationException
>   HResult=0x80131500
>   Message='Recommended pattern POST-201-LocationHeader' cannot be null.
>   Source=Microsoft.Rest.ClientRuntime.Azure
>   StackTrace:
>    at Microsoft.Rest.ClientRuntime.Azure.LRO.POSTLro`2.InitializeAsyncHeadersToUse()
>    at Microsoft.Rest.ClientRuntime.Azure.LRO.AzureLRO`2.<BeginLROAsync>d__15.MoveNext()
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at Microsoft.Rest.Azure.AzureClientExtensions.<GetLongRunningOperationResultAsync>d__1`2.MoveNext()
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at Microsoft.Rest.Azure.AzureClientExtensions.<GetPostOrDeleteOperationResultAsync>d__10`2.MoveNext()
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
>    at Microsoft.Azure.Management.Storage.FileSharesOperations.<LeaseWithHttpMessagesAsync>d__11.MoveNext() in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\src\Generated\FileSharesOperations.cs:line 1763
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
>    at Microsoft.Azure.Management.Storage.FileSharesOperationsExtensions.<LeaseAsync>d__13.MoveNext() in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\src\Generated\FileSharesOperationsExtensions.cs:line 530
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
>    at Microsoft.Azure.Management.Storage.FileSharesOperationsExtensions.Lease(IFileSharesOperations operations, String resourceGroupName, String accountName, String shareName, LeaseShareRequest parameters, String xMsSnapshot) in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\src\Generated\FileSharesOperationsExtensions.cs:line 493
>    at Storage.Tests.FileServiceTests.FileShareLease() in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\tests\Tests\FileServiceTests.cs:line 567

network trace:
```
POST https://management.azure.com/subscriptions/{subID}/resourceGroups/***/providers/Microsoft.Storage/storageAccounts/***/fileServices/default/shares/share9903/lease?api-version=2021-04-01 HTTP/1.1
x-ms-client-request-id: 28b2aab8-ba2f-42a8-b10e-11f6de493670
Accept-Language: en-US
Authorization: Bearer 
User-Agent: FxVersion/4.6.29916.01 OSName/Windows OSVersion/Microsoft.Windows.10.0.19042. Microsoft.Azure.Management.Storage.StorageManagementClient/21.2.0.0
Content-Type: application/json; charset=utf-8
Content-Length: 113
Host: management.azure.com

{
  "action": "Acquire",
  "leaseDuration": 60,
  "proposedLeaseId": "ca761232-ed42-11ce-bacd-************"
}
HTTP/1.1 201 Created
Cache-Control: no-cache
Pragma: no-cache
Content-Length: 50
Content-Type: application/json
Expires: -1
ETag: "0x8D9112F02094D0D"
x-ms-request-id: 4c03d176-d0c8-40fe-a542-f81b497ef3ae
Strict-Transport-Security: max-age=31536000; includeSubDomains
Server: Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0
x-ms-ratelimit-remaining-subscription-writes: 1199
x-ms-correlation-request-id: 429f70ec-582a-4f0c-9541-5d78e770d1d2
x-ms-routing-request-id: SOUTHEASTASIA:20210507T080635Z:429f70ec-582a-4f0c-9541-5d78e770d1d2
X-Content-Type-Options: nosniff
Date: Fri, 07 May 2021 08:06:35 GMT

{"leaseId":"ca761232-ed42-11ce-bacd-00aa0057b223"}
```

Reply:
For break lease, another error throw from SDK:

> Microsoft.Rest.ValidationException: 'Recommended patterns: POST-202-LocationHeader(Prefered)/AzureAsyncOperationHeader' cannot be null.
>    at Microsoft.Rest.ClientRuntime.Azure.LRO.POSTLro`2.InitializeAsyncHeadersToUse()
>    at Microsoft.Rest.ClientRuntime.Azure.LRO.AzureLRO`2.BeginLROAsync()
>    at Microsoft.Rest.Azure.AzureClientExtensions.GetLongRunningOperationResultAsync[TBody,THeader](IAzureClient client, AzureOperationResponse`2 response, Dictionary`2 customHeaders, CancellationToken cancellationToken)
>    at Microsoft.Rest.Azure.AzureClientExtensions.GetPostOrDeleteOperationResultAsync[TBody,THeader](IAzureClient client, AzureOperationResponse`2 response, Dictionary`2 customHeaders, CancellationToken cancellationToken)
>    at Microsoft.Azure.Management.Storage.FileSharesOperations.LeaseWithHttpMessagesAsync(String resourceGroupName, String accountName, String shareName, LeaseShareRequest parameters, String xMsSnapshot, Dictionary`2 customHeaders, CancellationToken cancellationToken) in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\src\Generated\FileSharesOperations.cs:line 1763
>    at Microsoft.Azure.Management.Storage.FileSharesOperationsExtensions.LeaseAsync(IFileSharesOperations operations, String resourceGroupName, String accountName, String shareName, LeaseShareRequest parameters, String xMsSnapshot, CancellationToken cancellationToken) in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\src\Generated\FileSharesOperationsExtensions.cs:line 530
>    at Microsoft.Azure.Management.Storage.FileSharesOperationsExtensions.Lease(IFileSharesOperations operations, String resourceGroupName, String accountName, String shareName, LeaseShareRequest parameters, String xMsSnapshot) in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\src\Generated\FileSharesOperationsExtensions.cs:line 493
>    at Storage.Tests.FileServiceTests.FileShareLease() in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\tests\Tests\FileServiceTests.cs:line 611

Reply:
Also see following exception for break Lease:

> Microsoft.Rest.ValidationException: 'Recommended patterns: POST-202-LocationHeader(Prefered)/AzureAsyncOperationHeader' cannot be null.
>    at Microsoft.Rest.ClientRuntime.Azure.LRO.POSTLro`2.InitializeAsyncHeadersToUse()
>    at Microsoft.Rest.ClientRuntime.Azure.LRO.AzureLRO`2.BeginLROAsync()
>    at Microsoft.Rest.Azure.AzureClientExtensions.GetLongRunningOperationResultAsync[TBody,THeader](IAzureClient client, AzureOperationResponse`2 response, Dictionary`2 customHeaders, CancellationToken cancellationToken)
>    at Microsoft.Rest.Azure.AzureClientExtensions.GetPostOrDeleteOperationResultAsync[TBody,THeader](IAzureClient client, AzureOperationResponse`2 response, Dictionary`2 customHeaders, CancellationToken cancellationToken)
>    at Microsoft.Azure.Management.Storage.FileSharesOperations.LeaseWithHttpMessagesAsync(String resourceGroupName, String accountName, String shareName, LeaseShareRequest parameters, String xMsSnapshot, Dictionary`2 customHeaders, CancellationToken cancellationToken) in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\src\Generated\FileSharesOperations.cs:line 1763
>    at Microsoft.Azure.Management.Storage.FileSharesOperationsExtensions.LeaseAsync(IFileSharesOperations operations, String resourceGroupName, String accountName, String shareName, LeaseShareRequest parameters, String xMsSnapshot, CancellationToken cancellationToken) in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\src\Generated\FileSharesOperationsExtensions.cs:line 530
>    at Microsoft.Azure.Management.Storage.FileSharesOperationsExtensions.Lease(IFileSharesOperations operations, String resourceGroupName, String accountName, String shareName, LeaseShareRequest parameters, String xMsSnapshot) in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\src\Generated\FileSharesOperationsExtensions.cs:line 493
>    at Storage.Tests.FileServiceTests.FileShareLease() in C:\code\srpsdk\sdk\storage\Microsoft.Azure.Management.Storage\tests\Tests\FileServiceTests.cs:line 611

Reply:
This is made sync now